go/parquet/encryption_read_config_test.go

```diff
@@ -0,0 +1,443 @@
+	}
+	return datadir
+}
+
+type TestDecryptionSuite struct {
+	suite.Suite
+
+	pathToDouble        string
+	pathToFloat         string
+	decryptionConfigs   []*parquet.FileDecryptionProperties
+	footerEncryptionKey string
+	colEncryptionKey1   string
+	colEncryptionKey2   string
+	fileName            string
+}
+
+func (d *TestDecryptionSuite) TearDownSuite() {
+	os.Remove(tempdir)
+}
+
+func TestFileEncryptionDecryption(t *testing.T) {
+	suite.Run(t, new(EncryptionConfigTestSuite))
+	suite.Run(t, new(TestDecryptionSuite))
+}
+
+func (d *TestDecryptionSuite) SetupSuite() {
+	d.pathToDouble = "double_field"
+	d.pathToFloat = "float_field"
+	d.footerEncryptionKey = FooterEncryptionKey
+	d.colEncryptionKey1 = ColumnEncryptionKey1
+	d.colEncryptionKey2 = ColumnEncryptionKey2
+	d.fileName = FileName
+
+	d.createDecryptionConfigs()
+}
+
+func (d *TestDecryptionSuite) createDecryptionConfigs() {
+	// Decryption configuration 1: Decrypt using key retriever callback that holds the
+	// keys of two encrypted columns and the footer key.
+	stringKr1 := make(encryption.StringKeyIDRetriever)
+	stringKr1.PutKey("kf", d.footerEncryptionKey)
+	stringKr1.PutKey("kc1", d.colEncryptionKey1)
+	stringKr1.PutKey("kc2", d.colEncryptionKey2)
+
+	d.decryptionConfigs = append(d.decryptionConfigs,
+		parquet.NewFileDecryptionProperties(parquet.WithKeyRetriever(stringKr1)))
+
+	// Decryption configuration 2: Decrypt using key retriever callback that holds the
+	// keys of two encrypted columns and the footer key. Supply aad_prefix.
+	stringKr2 := make(encryption.StringKeyIDRetriever)
+	stringKr2.PutKey("kf", d.footerEncryptionKey)
+	stringKr2.PutKey("kc1", d.colEncryptionKey1)
+	stringKr2.PutKey("kc2", d.colEncryptionKey2)
+	d.decryptionConfigs = append(d.decryptionConfigs,
+		parquet.NewFileDecryptionProperties(parquet.WithKeyRetriever(stringKr2), parquet.WithDecryptAadPrefix(d.fileName)))
+
+	// Decryption configuration 3: Decrypt using explicit column and footer keys. Supply
+	// aad_prefix.
+	decryptCols := make(parquet.ColumnPathToDecryptionPropsMap)
+	decryptCols[d.pathToFloat] = parquet.NewColumnDecryptionProperties(d.pathToFloat, parquet.WithDecryptKey(d.colEncryptionKey2))
+	decryptCols[d.pathToDouble] = parquet.NewColumnDecryptionProperties(d.pathToDouble, parquet.WithDecryptKey(d.colEncryptionKey1))
+	d.decryptionConfigs = append(d.decryptionConfigs,
+		parquet.NewFileDecryptionProperties(parquet.WithFooterKey(d.footerEncryptionKey), parquet.WithColumnKeys(decryptCols)))
+
+	// Decryption Configuration 4: use plaintext footer mode, read only footer + plaintext
+	// columns.
+	d.decryptionConfigs = append(d.decryptionConfigs, nil)
+}
+
+func (d *TestDecryptionSuite) decryptFile(filename string, decryptConfigNum int) {
+	// if we get decryption_config_num = x then it means the actual number is x+1
+	// and since we want decryption_config_num=4 we set the condition to 3
+	props := parquet.NewReaderProperties(memory.DefaultAllocator)
+	if decryptConfigNum != 3 {
```

Comment:
it might be more cleaner to encapsulate the test case into its own class/struct instead of the if/then here.  At the end of the day that might be just a slightly different form of obfuscation.